R/path.R

```diff
@@ -148,6 +140,8 @@ standardise_one_path <- function (path, write = FALSE) {
     stop("Can only read from, not write to, .zip", call. = FALSE)
   }
 
+  path <- enc2utf8(path)
```

Comment:
Why here? Is this because you use either `normalizePath()` or `check_path()` on the `path` above, and that might generate something in native encoding?

I would have assumed that if you expected the input `path` to already be UTF8, then you shouldn't have to call `enc2utf8()` again.

I also noticed that this `enc2utf8()` call is done _after_ the `path` would have been passed to the `archive::` functions (if we go through that code path). Is that okay?

Reply:
You've put your finger on something where I'm waving hands a bit.

Empirically, I am 100% certain that `enc2utf8()` is necessary in `standardise_one_path()`, in addition to `standardise_path()`, but I should probably review the exact sequence of calls that makes it so. This is part of the repeated calling back and forth from R to C++ back to R back to C++, etc.

As for the placement, I was unsure where to place it in the long block of code above dealing with archive files and writing files. So I placed it here, because it makes my existing tests pass. I expect this to get refined when I finish off this PR to account for file writing and, I suppose, reading compressed files.